javascripts/2048.js

```diff
@@ -1,32 +1,383 @@
 $(document).ready(function() {
   console.log('ready!');
+  initializeBoard();
   $('body').keydown(function(event){
     var arrow_keys = [37, 38, 39, 40];
     if(arrow_keys.indexOf(event.which) > -1) {
-      var tile = $('.tile');
-      moveTile(tile, event.which);
+      moveTile(event.which);
       event.preventDefault();
     }
   })
 })
 
-function moveTile(tile, direction) {
-  var new_tile_value = tile.attr("data-val") * 2;
-  tile.attr("data-val", new_tile_value);
-  tile.text(new_tile_value);
+var board     = [];
+var boardSize = 4;
+var newTile  = [];
+var score = 0;
 
+// sets up 2D array
+function initializeBoard() {
+  for(var i = 0; i < boardSize; i++) {
+    innerBoard = [];
+    board.push(innerBoard);
+
+    for(var j = 0; j < boardSize; j++) {
+      board[i][j] = 0;
+      }
```

Comment:
This indentation is strange and appears to throw off the rest of the indentation beyond here.